services/storage/virtualharddisk/client.go

```diff
@@ -74,3 +77,17 @@ func (c *VirtualHardDiskClient) Resize(ctx context.Context, group, container, na
 func (c *VirtualHardDiskClient) Precheck(ctx context.Context, group, container string, vhds []*storage.VirtualHardDisk) (bool, error) {
 	return c.internal.Precheck(ctx, group, container, vhds)
 }
+
+func (c *VirtualHardDiskClient) DownloadVhdFromHttp(ctx context.Context, group, container, name string, storage *storage.VirtualHardDisk, azHttpImg *compute.AzureGalleryImageProperties) (*storage.VirtualHardDisk, error) {
```

Comment:
Can we call it `UploadDiskFromHttp` to keep it consistent with the GalleryImage API.

Reply:
I think we should keep it DownloadVhdFromHttp as we are actually performing a download here and not Upload. Also we should rename them to download as well in GalleryImage API to avoid confusion in a seperate PR. I don't think writing upload is the correct way here. Let me know your thoughts.